tests/module/checks/test_git.py

```diff
@@ -0,0 +1,75 @@
+from pkgcheck.checks import git as git_mod
+from .. import misc
+from pkgcheck.git import _GitCommit
+
+def mk_commit(
+        message,
+        commit="7f9abd7ec2d079b1d0c36fc2f5d626ae0691757e",
+        author="author@domain.com", committer="committer@domain.com",
+        commit_date="Sun Dec 8 02:13:58 2019 -0700",
+    ):
+    return _GitCommit(commit, commit_date, author, committer, message)
+
+
+class TestGitCheck(misc.ReportTestCase):
+    check_kls = git_mod.GitCommitsCheck
+    check = git_mod.GitCommitsCheck(None)
+
+    @staticmethod
+    def commit(message, **kwargs):
+        """Helper function to create commits and cleanup message
+
+        Due to developer convenience, errant leading whitespace is
+        in the messages- this will strip that out so tags are properly
+        parsed."""
+        return mk_commit(tuple(x.lstrip() for x in message.splitlines()), **kwargs)
+
+    def test_sign_offs(self):
+        # assert that it checks for both author and comitter
+        r = self.assertReport(
+            self.check,
+            (self.commit("blah", author="user1", committer="user2"),)
+        )
+        assert isinstance(r, git_mod.MissingSignOff)
+        assert r.missing_sign_offs == ('user1', 'user2')
+
+        # assert that it handles author/committer being the same
+        self.assertNoReport(
+            self.check,
+            (self.commit("Signed-off-by: user@user.com",
```

Comment:
Remember that commit objects are just NamedTuples- this interacts poorly with the machinery in `misc.ReportTestCase`, hence having to do the manual tuple wrapping here.

One way to eliminate that- create actual classes for `_GitCommit`.  Doing so would also allow caching parsed tags- doing the parse once.  I'd probably implement it (since I'm usually API orientated), but it's your call if this is actually useful or not- I've been out of this codebase for many, many years.

Let me know your preference-

Reply:
Latest pkgcheck does use NamedTuple classes from the typing module for GitCommit. Does that help out in this case?

Reply:
Nope- it's still a tuple behind the scenes, the collections.NamedTuple (and types) is just wrappers of the class injecting some extra things.  types.NamedTuple is a nice change however.

Reply:
Actual class objs would be fine, I mostly used named tuples since I thought they'd be simpler and possibly help full git trees with grafted history get parsed faster on initial runs without resorting to dropping into CPython/cython.

However, I also haven't done any benchmarking to know how much the difference actually is, but I'd guess probably not that much in this case.